contracts/strategies/LPOptimiserStrategy.sol

```diff
@@ -43,135 +63,109 @@ contract LPOptimiserStrategy is DefaultAccessControl, ILpCallback {
     }
 
-    function settle() public {
+    function setTickValues(int24 currentTick, int24 tickLower, int24 tickUpper) public {
         _requireAtLeastOperator();
-
-        uint256[] memory tokenAmounts = new uint256[](1);
-        tokenAmounts[0] = type(uint256).max;
-
-        bytes memory options = abi.encode(
-                0,
-                0,
-                0,
-                false,
-                0,
-                0,
-                true,
-                1
-            );
-
-        _vault.pull(
-            address(_erc20Vault),
-            _tokens,
-            tokenAmounts,
-            options
-        );
+        _currentTick = currentTick;
+        _tickLower = tickLower;
+        _tickUpper = tickUpper;
     }
 
-    /// @notice Callback function called after for ERC20RootVault::deposit
-    function depositCallback() external {
-        update(_lastSignal, _lastLeverage, true);
+    /// @notice Get the current tick and position ticks and decide whether to rebalance
+    function rebalanceCheck() public view returns (bool) {
+        
+        // 1. Get current position, lower, and upper ticks
+        // _currentPosition = _vault.currentPosition();
+        // _tickLower = _currentPosition.tickLower;
+        // _tickUpper = _currentPosition.tickUpper;
+
+        // 2. Get current tick
+        // _currentTick = _periphery.getCurrentTick(_marginEngine);
+
+        // 3. Compare current fixed rate to lower and upper bounds
+        if (
+            _tickLower - _logProximity <= _currentTick &&
+            _currentTick <= _tickUpper + _logProximity
+            
+            ) {
+            // 4.1. If current fixed rate is within bounds, return false (don't rebalance)
+            return false;
+        } else {
+            // 4.2. If current fixed rate is outside bounds, return true (do rebalance)
+            return true;
+        }
     }
 
-    /// @notice Callback function called after for ERC20RootVault::withdraw
-    function withdrawCallback() external {
-        update(_lastSignal, _lastLeverage, true);
+    /// @notice Get the nearest tick multiple given a tick and tick spacing
+    function nearestTickMultiple(int24 newTick, int24 tickSpacing) public pure returns (int24) {
+     return (newTick / tickSpacing + ( ( ((newTick % tickSpacing) + tickSpacing) % tickSpacing ) >= tickSpacing/2 ? int24(1) : int24(0)) ) * tickSpacing;
     }
+
+    /// @notice Set new optimimal tick range based on current tick
+    /// @param currentFixedRateWad currentFixedRate which is passed in from a 7-day rolling avg. historical fixed rate.
+    function rebalance (uint256 currentFixedRateWad) public returns (int24 newTickLowerMul, int24 newTickUpperMul) {
+        _requireAtLeastOperator();
+
+            // 0. Get tickspacing from vamm
+            // _tickSpacing = _vamm.tickSpacing(_vamm);
+
+            // 1. Get the new tick lower
+            int256 deltaWad = int256(currentFixedRateWad - _sigmaWad);
```

Comment:
Edge case here: What if _sigmaWad > currentFixedRateWad? It underflows in this situation. You should consider int256(currentFixedRateWad) - int256(_sigmaWad).

You should write a test that captures this scenario.

Reply:
Modified this and added tests